src/test/java/org/broadinstitute/hellbender/tools/walkers/bqsr/ApplyBQSRDataflowIntegrationTest.java

```diff
@@ -0,0 +1,78 @@
+package org.broadinstitute.hellbender.tools.walkers.bqsr;
+
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.IntegrationTestSpec;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public final class ApplyBQSRDataflowIntegrationTest extends CommandLineProgramTest {
+    private static class ABQSRTest {
+        final String bam;
+        final String args;
+        final String expectedFile;
+
+        private ABQSRTest(String bam, String args, String expectedFile) {
+            this.bam= bam;
+            this.args = args;
+            this.expectedFile = expectedFile;
+        }
+
+        @Override
+        public String toString() {
+            return String.format("ApplyBQSR(args='%s')", args);
+        }
+    }
+
+    final String resourceDir = getTestDataDir() + "/" + "BQSR" + "/";
+    final String hiSeqBam = resourceDir + "HiSeq.1mb.1RG.2k_lines.alternate.bam";
+
+    @DataProvider(name = "ApplyBQSRTest")
+    public Object[][] createABQSRTestData() {
+        List<Object[]> tests = new ArrayList<>();
+
+        tests.add(new Object[]{new ABQSRTest(hiSeqBam, "", resourceDir + "expected.HiSeq.1mb.1RG.2k_lines.bqsr.alternate.bam")});
+        tests.add(new Object[]{new ABQSRTest(hiSeqBam, " -qq -1", resourceDir + "expected.HiSeq.1mb.1RG.2k_lines.bqsr.qq-1.alternate.bam")});
+        tests.add(new Object[]{new ABQSRTest(hiSeqBam, " -qq 6", resourceDir + "expected.HiSeq.1mb.1RG.2k_lines.bqsr.qq6.alternate.bam")});
+        tests.add(new Object[]{new ABQSRTest(hiSeqBam, " -DIQ", resourceDir + "expected.HiSeq.1mb.1RG.2k_lines.bqsr.DIQ.alternate.bam")});
+
+        return tests.toArray(new Object[][]{});
+    }
+
+    @Test(dataProvider = "ApplyBQSRTest")
+    public void testPR(ABQSRTest params) throws IOException {
+        IntegrationTestSpec spec = new IntegrationTestSpec(
+                        " -I " + params.bam +
+                        " --bqsr_recal_file " + resourceDir + "HiSeq.20mb.1RG.table.gz" +
+                        params.args +
+                        " -O %s",
```

Comment:
We should explicitly specify the test runner using `addDataflowRunnerArgs(args)` from BaseTest, this is a new thing to enable us to run tests seamlessly on spark as well as dataflow.  It looks like `addDataflowRunner` doesn't slot in nicely to these integration test configurations, so we should probably add new function getDataflow() that just returns the string needed.